packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -146,9 +146,9 @@ contract MetaTransactionWallet is
   function getMetaTransactionDigest(
     address destination,
     uint256 value,
-    bytes memory data,
+    bytes calldata data,
```

Comment:
Out of curiosity what's the consequence of this? Is it cheaper to keep it as calldata? Would one set it as memory if it will end up in storage?

Reply:
Good question. First of all, it is *necessary* to change it to `calldata` - that's the only data area for reference type parameters to `external` functions. This is also the reason why `external` functions are cheaper than `public` - you can't use `calldata` in a public function, would have to use `memory` in this case, which would mean, when calling externally, we additionally need to copy from the calldata to memory.

[Here's](https://docs.soliditylang.org/en/v0.5.13/types.html#reference-types) the docs on data locations.

Reply:
Makes sense, thanks for the brain dump!